laplace/baselaplace.py

```diff
@@ -846,6 +846,10 @@ def fit(
             else:
                 X, y = data
                 X, y = X.to(self._device), y.to(self._device)
+
+            if self.likelihood == Likelihood.REGRESSION and y.ndim == out.ndim - 1:
+                y = y.unsqueeze(-1)
```

Comment:
Should we also raise a warning in this case? It might be that the model was trained the wrong way because the standard MSELoss also assumes that `y` and `out` have the same dimensionality.

Reply:
Thanks, I think raising an error is the way to go. Updated!